hack/test/e2e-image-cache.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -eoux pipefail
+
+source ./hack/test/e2e.sh
+
+PROVISIONER=qemu
+CLUSTER_NAME=e2e-image-cache
+
+NODE="172.20.2.2"
+
+function build_special_mirrors {
+  REGISTRY_MIRROR_FLAGS=()
+
+  for registry in docker.io quay.io ghcr.io; do
```

Comment:
the cache should also work for the installer?

Reply:
I actually don't know how to do in portable way:

* The known part is updating `cache-create` stage: new command should look something like `@( $(TALOSCTL_EXECUTABLE) images default | grep -v 'siderolabs'; echo "$(REGISTRY_AND_USERNAME)/installer:$(IMAGE_TAG)" ) | $(TALOSCTL_EXECUTABLE) images cache-create --image-cache-path=/tmp/cache.tar --images=- --force`
* The unknown part is how to create `image-cache.yaml` where one of the registries names is dependent on `${IMAGE_REGISTRY}` variable.

Reply:
```
images default | grep -v installer > image.txt
echo ${IMAGE_REGISTRY}/siderolabs/installer:${TAG} >> images.txt
cat images.txt | <cache create>
```

Reply:
Thats already done. What I'm saying is I have the patch file

```yaml
machine:
  features:
    imageCache:
      localEnabled: true
  registries:
    mirrors:
      gcr.io:
        skipFallback: true
        endpoints:
        - http://172.20.0.251:65000
      k8s.gcr.io:
        skipFallback: true
        endpoints:
        - http://172.20.0.251:65000
      registry.k8s.io:
        skipFallback: true
        endpoints:
        - http://172.20.0.251:65000
      <registry-name-should-be-inserted-during-ci-step>:
        skipFallback: true
        endpoints:
        - http://172.20.0.251:65000
---
apiVersion: v1alpha1
kind: VolumeConfig
name: IMAGECACHE
provisioning:
  diskSelector:
    match: 'system_disk'
```

Reply:
ahh can't we use `*` host and add `skipFallback` for it, so it catches all, or just `yq` it in like other steps

Reply:
@frezbo yeah, it worked. Now we only download stuff from the cache.